packages/gateway/src/manager.ts

```diff
@@ -54,6 +55,160 @@ export function createGatewayManager(options: CreateGatewayManagerOptions): Gate
+        // Handles preparing mid sized bots for LBS
+        gateway.totalShards = gateway.calculateTotalShards()
+        // Set first shard id if provided in info
+        if (info.firstShardId) gateway.firstShardId = info.firstShardId
+        // Set last shard id if provided in info
+        if (info.lastShardId) gateway.lastShardId = info.lastShardId
+        logger.warn(`[Resharding] Starting the reshard process. New Total Shards. ${gateway.totalShards}`)
+
+        // Resetting buckets
+        gateway.buckets.clear()
+        // Refilling buckets with new values
+        gateway.prepareBuckets()
+
+        // SPREAD THIS OUT TO DIFFERENT WORKERS TO BEGIN STARTING UP
+        gateway.buckets.forEach(async (bucket, bucketId) => {
+          for (const worker of bucket.workers) {
+            for (const shardId of worker.queue) {
+              await gateway.resharding.tellWorkerToPrepare(worker.id, shardId, bucketId)
+            }
+          }
+        })
+      },
+      async tellWorkerToPrepare(workerId, shardId, bucketId) {
+        logger.warn(`[Resharding] Telling worker to prepare. Worker: ${workerId} | Shard: ${shardId} | Bucket: ${bucketId}`)
+        const shard = new Shard({
+          id: shardId,
+          connection: {
+            compress: gateway.compress,
+            intents: gateway.intents,
+            properties: gateway.properties,
+            token: gateway.token,
+            totalShards: gateway.totalShards,
+            url: gateway.url,
+            version: gateway.version,
+          },
+          // Ignore events until we are ready
+          events: {
+            async message(shard, payload) {
+              if (payload.t === 'READY') {
+                await gateway.resharding.updateGuildsShardId(
+                  (payload.d as DiscordReady).guilds.map((g) => g.id),
+                  shardId,
+                )
+              }
+            },
+          },
+          requestIdentify: async () => {
+            await gateway.identify(shardId)
+          },
+          shardIsReady: async () => {
+            logger.debug(`[Shard] Shard #${shardId} is ready`)
+            await delay(gateway.spawnShardDelay)
+            logger.debug(`[Shard] Resolving shard identify request`)
+            gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)!.identifyRequests.shift()?.()
+          },
+        })
+
+        if (gateway.preferSnakeCase) {
+          shard.forwardToBot = async (payload) => {
+            options.events?.message?.(shard, payload)
+          }
+        }
+
+        gateway.resharding.shards.set(shardId, shard)
+
+        const bucket = gateway.buckets.get(shardId % gateway.connection.sessionStartLimit.maxConcurrency)
+        if (!bucket) return
+
+        return await new Promise((resolve) => {
+          // Mark that we are making an identify request so another is not made.
+          bucket.identifyRequests.push(resolve)
```

Comment:
I can't seem to find any lines checking that "if we are making an identify request, don't make another one" or something like that?

Reply:
If you mean a second request on the same shard it's handled by the queue. However, if you're talking about multiple shards then sometimes you need to allow that like bigger ones that have a higher Max concurrency